zenodo/modules/deposit/utils.py

```diff
@@ -219,6 +220,17 @@ def delete_record(record_uuid, reason, user):
         openaire_delete.delay(original_id=original_id,
                               datasource_id=datasource_id)
 
+    # Delete access requests as well as access links
```

Comment:
Hm, can this be moved higher up to avoid doing `db.session.commit()` multiple times? E.g. before [this line](https://github.com/zenodo/zenodo/pull/2280/files#diff-fff270fd242463c8a79d99f4e6f0c0d4b6940339080c2cef5cb94e25d2519530R204).

Reply:
Tested and done :)